cmd/di.go

```diff
@@ -329,10 +331,22 @@ func (di *Dependencies) bootstrapNetworkComponents(options node.OptionsNetwork)
 		network.EtherClientRPC = options.EtherClientRPC
 	}
 
+	if options.StatisticsAddress != metadata.DefaultNetwork.ELKAddress {
+		network.ELKAddress = options.StatisticsAddress
+	}
+
 	di.NetworkDefinition = network
 	di.MysteriumAPI = mysterium.NewClient(network.DiscoveryAPIAddress)
 	di.MysteriumMorqaClient = oracle.NewMorqaClient(network.QualityOracle)
 
+	var eventsTransport events.Transport
+	if options.DisableStatistics {
+		eventsTransport = events.NewNilTransport()
+	} else {
+		eventsTransport = events.NewELKTransport(network.ELKAddress, 10*time.Second)
+	}
+	di.EventsSender = &events.Sender{Transport: eventsTransport}
```

Comment:
could also create a sub package in the events(or preferably the metrics) package, call it a factory and just implement the whole block there. Here we'd only have a single call to the factory with params and we would not make the DI any more heavy than it is.

Reply:
Extracted this as `metrics.CreateSender`